File: Models/Data/Atributes/Country.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace ProductsReviewsAngular.Models
{
    public class Country
    {
        public int idCountry { get; set; }
        [ConcurrencyCheck]
        [MaxLength(50)]
        public string name { get; set; }
        public string normalizedName { get; set; }
        public virtual ICollection<Producer> Producers { get; set; }
    }
}

File: ClientApp/src/app/models/Country.ts
import {Producer} from "./Producer";
export class Country{    
    public idCountry?: number;
    public name?: string;
    public Producers?: Producer[];

    constructor(params: Country = {} as Country){
        let{
            idCountry = undefined,
            name = undefined,
            Producers = []
        } = params;
        this.idCountry = idCountry;
        this.name = name;
        this.Producers = Producers;
    }
}